### PR TITLE
Only propagating updated_at if records have been changed with touch option

### PR DESCRIPTION
**Pull request by brocktimus**
This is a sister PR to the main rails one I've written rails/rails#14979. 

Basically, just adjusting the callback to only run if the record has actually changed. This brings it into line with updated_at on the main record itself which only updates if the record changed.

### CODE REVIEW

File: lib/ancestry/has_ancestry.rb [hunk 1]
```diff
@@ -74,9 +74,9 @@ def has_ancestry options = {}
       }
     end
 
-    after_save :touch_ancestors_callback
     after_touch :touch_ancestors_callback
     after_destroy :touch_ancestors_callback
```
Comment (mch83) [inline lib/ancestry/has_ancestry.rb]: test comment

File: test/concerns/touching_test.rb [hunk 2]
```diff
@@ -44,4 +44,25 @@ def test_touch_option_enabled
       assert_equal way_back, child_1_2.reload.updated_at,        "unrelated record was touched"
     end
   end
-end
\ No newline at end of file
+
```
Comment (mch83) [inline test/concerns/touching_test.rb]: test

**Comment by kbrock** [comment]
@stefankroes this looks good to me. Any reason to hold off?

**Comment by stefankroes** [comment]
Nope, should be an improvement

**Comment by brocktimus** [comment]
@kbrock noticed you're now a collaborator so just bumping this a little? (Sorry, not wanting to nag).

**Pull request merged**